projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts

```diff
@@ -319,19 +335,19 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
       return;
     }
 
-    new CartesianNoDataMessage(this.chartBackgroundSvgElement, this.allSeries).updateMessage();
+    new CartesianNoDataMessage(this.chartBackgroundSvgElement, this.series).updateMessage();
   }
 
   private drawLegend(): void {
     if (this.chartContainerElement) {
       if (this.legendPosition !== undefined && this.legendPosition !== LegendPosition.None) {
-        this.legend = new CartesianLegend(this.allSeries, this.injector, this.intervalData, this.summaries).draw(
+        this.legend = new CartesianLegend(this.series, this.injector, this.intervalData, this.seriesSummaries).draw(
```

Comment:
Should the range be part of the legend?

Reply:
Going to circle back to legend and tooltip separately.